synapse/handlers/message.py

```diff
@@ -91,7 +91,7 @@ async def get_room_data(
         room_id: str,
         event_type: str,
         state_key: str,
-    ) -> dict:
+    ) -> Optional[EventBase]:
```

Comment:
This seems to return `EventBase`, and the caller calls methods on it that don't exist on `dict`. 🤷